examples/vision/md/supervised-contrastive-learning.md

```diff
@@ -8,8 +8,28 @@
 
 <img class="k-inline-icon" src="https://colab.research.google.com/img/colab_favicon.ico"/> [**View in Colab**](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/vision/ipynb/supervised-contrastive-learning.ipynb)  <span class="k-dot">•</span><img class="k-inline-icon" src="https://github.com/favicon.ico"/> [**GitHub source**](https://github.com/keras-team/keras-io/blob/master/examples/vision/supervised-contrastive-learning.py)
 
+## Introduction
 
+[Supervised Contrastive Learning](https://arxiv.org/abs/2004.11362)
+(Prannay Khosla et al.) is a training methodology that outperforms
+plain crossentropy-supervised training on classification tasks.
```

Comment:
Hey @ksalama I have a few proposals to improve this intro, if you don't mind.

I think here you're describing the results of an experiment (by saying it "outperforms"). Maybe it'd be more useful for the readers to first learn about the gist of this supervised contrastive learning (SCL) and how it works in 1-2 sentences. 

Then, you could finish off this small introductory paragraph with the "outperforms" statement, while also being explicit about how SCL outperforms the traditional vanilla cross-entropy supervised learning stuff (judging by Table 2 and 3 on page 7 of https://arxiv.org/pdf/2004.11362.pdf, SCL "outperforms" i.t.o. accuracy by a margin).

I think the cool thing about SCL is that it extends the previous (?) self-supervised approach to supervised learning - you should definitely highlight it here (_"the self-supervised batch contrastive approach to the fully-supervised"_ - page 1, https://arxiv.org/pdf/2004.11362.pdf). SCL "contrasts the set of all samples from the same class as positives against the negatives from the remainder of the batch" (page 2, figure 3).

On page 4 under "Method", the paper actually summarizes what SCL does:

_"Given an input batch of data, we first apply data augmentation twice to obtain two copies of the batch. Both copies are forward propagated through the encoder network to obtain a 2048-dimensional normalized embedding. During training, this representation is further propagated through a projection network that is discarded at inference time. The supervised contrastive loss is computed on the outputs of the projection network. To use the trained model for classification, we train a linear classifier on top of the frozen representations using a cross-entropy loss."_

I recommend you summarize it in a human-friendly way to appeal to non-academics.

I can assist you with that, if you need help.

Anyway, these are just suggestions.

Reply:
@8bitmp3 Thanks a lot for suggestion. Please feel free to provide a intro text that you think it could be simple and useful, and will be happy to commit your suggestion

Reply:
@ksalama Would it fair to say that the the supervised contrastive learning paper introduces a way of training that includes the supervised contrastive loss? Also, we could say that the method offers a two-stage framework that enhances the image classification performance (borrowed from: https://github.com/sayakpaul/Supervised-Contrastive-Learning-in-TensorFlow-2). I also like how they worded it here: _"Learn how to map the normalized encoding of samples belonging to the same category closer and the samples belonging to the other classes farther."_ (https://wandb.ai/authors/scl/reports/Improving-Image-Classifiers-With-Supervised-Contrastive-Learning--VmlldzoxMzQwNzE). We could rephrase it with attribution. cc @fchollet

Reply:
Bear with me, there's a talk by one of the sponsors at the NeurIPS conference today - and probably next week by the paper's authros - that cover(s) contrastive and supervised contrastive learning. I'll take some notes and revise the intro to make it more useful for the readers. cc @fchollet

Reply:
@8bitmp3 Sounds good. I would merge this basic introduction to the `.md` file so that the example page on the website will have "an" introduction (as it is currently hasn't!). Then we can update the introduction as you suggest.